# Houston Air Quality Data Exploration
____________________
### Patrick Huston, Ezra Varday, Charlie Owen | FA 2017

This notebook presents an initial exploration into the EPA air monitoring data for the Park Place AQ monitoring site in Houston, Texas.

##### Motivating Questions:
- What does air quality look like on an average day in Houston? 
- What are the seasonal dynamics of AQ in Houston? 
- Etc.

In [2]:
%matplotlib inline

import numpy as np
import pandas as pd

In [5]:
!ls ../data

338120170929155414557.txt 338120170929160847395.txt


In [6]:
filename = "../data/338120170929155414557.txt"

In [11]:
df1 = pd.read_csv(filename, skiprows=10, sep='|', usecols=range(13), parse_dates={"Datetime" : [10,11]})
df1.tail()

Datetime Transaction Type Action  State Cd  County Cd  \
339573 2016-12-30 19:00:00               RD      I        48        201   
339574 2016-12-30 20:00:00               RD      I        48        201   
339575 2016-12-30 21:00:00               RD      I        48        201   
339576 2016-12-30 22:00:00               RD      I        48        201   
339577 2016-12-30 23:00:00               RD      I        48        201   

        Site ID  Parameter Cd  POC  Dur Cd  Unit Cd  Meth Cd    Value  
339573      416         64101    1       1       16       11  1020.89  
339574      416         64101    1       1       16       11  1020.47  
339575      416         64101    1       1       16       11  1019.77  
339576      416         64101    1       1       16       11  1019.07  
339577      416         64101    1       1       16       11  1018.31

In [25]:
gb = df1.groupby('Parameter Cd')

gb['Unit Cd'].describe()

count  mean  std   min   25%   50%   75%   max
Parameter Cd                                                  
42101          8555.0   7.0  0.0   7.0   7.0   7.0   7.0   7.0
42601         25170.0   8.0  0.0   8.0   8.0   8.0   8.0   8.0
42602         25127.0   8.0  0.0   8.0   8.0   8.0   8.0   8.0
42603         25170.0   8.0  0.0   8.0   8.0   8.0   8.0   8.0
44201         25466.0   8.0  0.0   8.0   8.0   8.0   8.0   8.0
61101         25291.0  12.0  0.0  12.0  12.0  12.0  12.0  12.0
61103         25289.0  12.0  0.0  12.0  12.0  12.0  12.0  12.0
61104         25289.0  14.0  0.0  14.0  14.0  14.0  14.0  14.0
61106         25289.0  14.0  0.0  14.0  14.0  14.0  14.0  14.0
62101         25871.0  15.0  0.0  15.0  15.0  15.0  15.0  15.0
62103         25513.0  15.0  0.0  15.0  15.0  15.0  15.0  15.0
62201         25689.0  19.0  0.0  19.0  19.0  19.0  19.0  19.0
63301         25983.0  25.0  0.0  25.0  25.0  25.0  25.0  25.0
64101         25876.0  16.0  0.0  16.0  16.0  16.0  16.0  16.0

### Parameter Breakdown

| Parameter Code | Meaning                    | Unit Code | Unit        |
|----------------|----------------------------|-----------|-------------|
| 42101          | Carbon dioxide             | 7         | ppm         |
| 42601          | Nitric oxide               | 8         | ppb         |
| 42602          | Nitrogen dioxide           | 8         | ppb         |
| 42603          | NOx                        | 8         | ppb         |
| 42201          | Ozone                      | 8         | ppb         |
| 61101          | Wind spd - scalar          | 12        | mph         |
| 61103          | Wind speed - resultant     | 12        | mph         |
| 61104          | Wind direction - resultant | 14        | deg comp.   |
| 61106          | Std. dev hz wind direction | 14        | deg comp.   |
| 62101          | Outdoor temperature        | 15        | Deg. F      |
| 62103          | Dew point                  | 15        | Deg. F      |
| 62201          | Humidity                   | 19        | % rel. hum  |
| 63301          | Solar radiation            | 25        | Lang/minute |
| 64101          | Barometric pressure        | 16        | Millibars   |